src/map.rs

```diff
@@ -532,6 +532,198 @@ where
+                                //  - another thread already has a reference to now_garbage.
+                                //    they must have read it before the call to swap.
+                                //    because of this, that thread must be pinned to an epoch <=
+                                //    the epoch of our guard. since the garbage is placed in our
+                                //    epoch, it won't be freed until the _next_ epoch, at which
+                                //    point, that thread must have dropped its guard, and with it,
+                                //    any reference to the value.
+                                //  - another thread is about to get a reference to this value.
+                                //    they execute _after_ the swap, and therefore do _not_ get a
+                                //    reference to now_garbage (they get value instead). there are
+                                //    no other ways to get to a value except through its Node's
+                                //    `value` field (which is what we swapped), so freeing
+                                //    now_garbage is fine.
+                                unsafe { guard.defer_destroy(now_garbage) };
+                            } else {
+                                delta = -1;
+                                // remove the BinEntry containing the removed key value pair from the bucket
+                                if !pred.is_null() {
+                                    // either by changing the pointer of the previous BinEntry, if present
+                                    // safety: see remove
+                                    unsafe { pred.deref() }
+                                        .as_node()
+                                        .unwrap()
+                                        .next
+                                        .store(next, Ordering::SeqCst);
+                                } else {
+                                    // or by setting the next node as the first BinEntry if there is no previous entry
+                                    t.store_bin(bini, next);
+                                }
+
+                                // in either case, mark the BinEntry as garbage, since it was just removed
+                                // safety: see remove
+                                unsafe { guard.defer_destroy(p) };
+                            }
+                            break Some(current_value);
+                        }
+
+                        pred = p;
+                        if next.is_null() {
+                            // we're at the end of the bin
+                            break None;
+                        }
+                        p = next;
+
+                        bin_count += 1;
+                    };
+                    drop(head_lock);
+
+                    // TODO: TREEIFY_THRESHOLD
```

Comment:
It doesn't seem like the Java code considers treeifying in `compute_if_present`, so this comment can go away.